packages/flutter_tools/test/commands/precache_test.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/common.dart';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/precache.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+import 'package:flutter_tools/src/version.dart';
+import 'package:mockito/mockito.dart';
+
+import '../src/common.dart';
+import '../src/context.dart';
+import '../src/mocks.dart';
+
+void main() {
+  final MockDeviceManager mockDeviceManager = MockDeviceManager();
+
+  group('precache', () {
+    final MockCache cache = MockCache();
+    final MockFlutterVersion flutterVersion = MockFlutterVersion();
+   
+    Set<DevelopmentArtifact> artifacts;
+
+    when(cache.isUpToDate()).thenReturn(false);
+    when(cache.updateAll(any)).thenAnswer((Invocation invocation) {
+      artifacts = invocation.positionalArguments.first;
+      return Future.value(null);
+    });
+    when(flutterVersion.isStable).thenReturn(true);
```

Comment:
It took me a couple of reads to realize that this mock was only applied in the second test's overrides; I was confused at first to how the test was passing. Maybe move the declaration and setup of `flutterVersion` down to between the two tests to make it clearer where it's used?